homeassistant/components/tplink/switch.py

```diff
@@ -35,70 +26,38 @@ async def async_setup_entry(
+    coordinators: dict[str, TPLinkDataUpdateCoordinator] = hass.data[TPLINK_DOMAIN][
         COORDINATORS
     ]
-    switches: list[SmartPlug] = hass.data[TPLINK_DOMAIN][CONF_SWITCH]
+    switches: list[SmartDevice] = hass.data[TPLINK_DOMAIN][CONF_SWITCH]
     for switch in switches:
-        coordinator = coordinators[switch.context or switch.mac]
+        coordinator = coordinators[switch.device_id]
         entities.append(SmartPlugSwitch(switch, coordinator))
+        if switch.is_strip:
+            _LOGGER.info("initializing strip with %s sockets", len(switch.children))
+            for child in switch.children:
+                entities.append(SmartPlugSwitch(child, coordinator))
 
     async_add_entities(entities)
 
 
-class SmartPlugSwitch(CoordinatorEntity, SwitchEntity):
+class SmartPlugSwitch(CoordinatedTPLinkEntity, SwitchEntity):
     """Representation of a TPLink Smart Plug switch."""
 
-    def __init__(
-        self, smartplug: SmartPlug, coordinator: DataUpdateCoordinator
-    ) -> None:
-        """Initialize the switch."""
-        super().__init__(coordinator)
-        self.smartplug = smartplug
-
-    @property
-    def data(self) -> dict[str, Any]:
-        """Return data from DataUpdateCoordinator."""
-        return self.coordinator.data
-
-    @property
-    def unique_id(self) -> str | None:
-        """Return a unique ID."""
-        return self.data[CONF_DEVICE_ID]
-
-    @property
-    def name(self) -> str | None:
-        """Return the name of the Smart Plug."""
-        return self.data[CONF_ALIAS]
-
-    @property
-    def device_info(self) -> DeviceInfo:
-        """Return information about the device."""
-        return {
-            "name": self.data[CONF_ALIAS],
-            "model": self.data[CONF_MODEL],
-            "manufacturer": "TP-Link",
-            "connections": {(dr.CONNECTION_NETWORK_MAC, self.data[CONF_MAC])},
-            "sw_version": self.data[CONF_SW_VERSION],
-        }
-
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if switch is on."""
-        return self.data[CONF_STATE]
-
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the switch on."""
-        await self.hass.async_add_executor_job(self.smartplug.turn_on)
+        await self.device.turn_on()
         # Workaround for delayed device state update on HS210: #55190
-        if "HS210" in self.device_info["model"]:
+        if "HS210" in self.device.model:
             await sleep(0.5)
+
         await self.coordinator.async_refresh()
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the switch off."""
-        await self.hass.async_add_executor_job(self.smartplug.turn_off)
+        await self.device.turn_off()
         # Workaround for delayed device state update on HS210: #55190
-        if "HS210" in self.device_info["model"]:
+        if "HS210" in self.device.model:
             await sleep(0.5)
+
         await self.coordinator.async_refresh()
```

Comment:
If its a smart strip we don't want to refresh the children here

Reply:
We should be able to get the new status back from the request so we shouldn't need a refresh

Reply:
We could set a property on the coordinator so the next update doesn't update the children.

Reply:
Using the response from the device could be used as an optimization, but some devices (iirc specific bulbs) do not report their new state. We could still make an assumption that the state change succeeded and let the next update cycle to handle any disparities.

We could also shorten the update cycle from the current 30 seconds to something like five seconds, especially after your newest changes on re-using the existing connection.